Source/VCExpr/ScopedNamer.cs

```diff
@@ -174,7 +174,11 @@ public virtual string GetName(object thing, string inherentName)
       }
 
       var uniqueInherentName = NextFreeName(thing, inherentName);
-      if (boogieDeterminedNames.Contains(inherentName))
+      if (thing is DatatypeConstructor || thing is DatatypeMembership || thing is DatatypeSelector)
```

Comment:
This seems hard to maintain since it's adding specific code in a generic location. New AST constructs could easily forget to update this `if` if they need to be added. I'd love to suggest something but I can't without understanding well why these three should be here.